lib/chef/knife/bootstrap/client_builder.rb

```diff
@@ -138,7 +141,7 @@ def node
               node = Chef::Node.new(chef_server_rest: client_rest)
               node.name(node_name)
               node.run_list(normalized_run_list)
-              node.normal_attrs = first_boot_attributes if first_boot_attributes
+              node.normal_attrs = first_boot_attributes if first_boot_attributes # TODO: should environment be saved here, if present?
```

Comment:
I think the answer to this TODO is `no` because the node will be assigned an environment.  And nothing about these normal attributes communicate that they are first_boot_attributes unless you know the implementation of this.